instance/models/mixins/openedx_theme.py

```diff
@@ -115,14 +114,48 @@ def get_theme_settings(self):
             ],
             "SIMPLETHEME_ENABLE_DEPLOY": True,
             "EDXAPP_DEFAULT_SITE_THEME": "simple-theme",
+            "EDXAPP_COMPREHENSIVE_THEME_SOURCE_REPO": settings.EDXAPP_COMPREHENSIVE_THEME_SOURCE_REPO,
+            "EDXAPP_COMPREHENSIVE_THEME_VERSION": settings.EDXAPP_COMPREHENSIVE_THEME_VERSION,
             "SIMPLETHEME_EXTRA_SASS": """
-                .global-header {{
-                    background: {header_bg};
-                }}
-                .wrapper-footer {{
-                    background: {footer_bg};
-                }}""".format(header_bg=application.header_bg_color,
-                             footer_bg=application.footer_bg_color)
+                $main-color: {main_color};
+                $link-color: {link_color};
+                $header-bg: {header_bg};
+                $header-font-color: {header_font_color};
+                $footer-bg: {footer_bg};
+                $footer-font-color: {footer_font_color};
+            """.format(
+                link_color=application.link_color,
+                main_color=application.main_color,
+                header_bg=application.header_bg_color,
+                header_font_color=self.get_contrasting_font_color(
+                    application.header_bg_color
+                ),
+                footer_bg=application.footer_bg_color,
+                footer_font_color=self.get_contrasting_font_color(
+                    application.footer_bg_color
+                ),
+            )
         }
 
-        return yaml.dump(settings, default_flow_style=False)
+        return yaml.dump(theme_settings, default_flow_style=False)
+
+    @staticmethod
+    def get_contrasting_font_color(background_color, delta=0.5):
```

Comment:
@giovannicimolin, can you write some tests for this method just to make sure it gives the expected results for a few colors?

Reply:
Sure, done!